### PR TITLE
Implementation of Trapezoidal Rule using Python3 and Numpy

### LINKED ISSUE #16: Add Trapezoidal Rule Implementation
# Description

Implement the ```Trapezoidal rule``` in ```Python``` with ```Numpy``` to compute the area of between a function f(x) and the x-axis.
- Add implementation
- Add some tests

### PR DESCRIPTION
**Pull request by Dilpreet047**
Output for the function trapz(np.sin,0,np.pi/2,1000)
0.9999997943832332
Output for the function trapz(lambda x : 3*x**2,0,1,10000)
1.0000000050000002

Closes #16

### CODE REVIEW

**Comment by gmuraru** [comment]
Could you also add some tests :D

File: Math/Approximations/TrapezoidRule/Implementation.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import numpy as np
```
Comment (gmuraru) [inline Math/Approximations/TrapezoidRule/Implementation.py:1]: Leave 2 empty spaces after this

File: Math/Approximations/TrapezoidRule/Implementation.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    '''The trapezoid rule approximates the integral int_a^b f(x) dx by the sum:
+    (dx/2)/sum_{k=1}^N (f(x_k) + f(x_{k-1}))
+    where x_k = a + k*dx and dx = (b - a)/N.
+    '''
+    x = np.linspace(a,b,N+1) # N+1 points make N subintervals
+    y = f(x)
+    y_right = y[1:] # right endpoints
+    y_left = y[:-1] # left endpoints
+    dx = (b - a)/N
+    T = (dx/2) * np.sum(y_right + y_left)
```
Comment (gmuraru) [inline Math/Approximations/TrapezoidRule/Implementation.py]: Q: Could you use small ```t``` here. I think ```T``` is reserved for type hints?

File: Math/Approximations/TrapezoidRule/Implementation.py (hunk 1 above)
Comment (Dilpreet047) [reply Math/Approximations/TrapezoidRule/Implementation.py:1]: Done.

File: Math/Approximations/TrapezoidRule/Implementation.py (hunk 2 above)
Comment (Dilpreet047) [reply Math/Approximations/TrapezoidRule/Implementation.py]: Done

**Comment by gmuraru** [comment]
Also, you need to add some tests (look at the test folder) + the commits should respect a given format.

**Review by gmuraru — CHANGES_REQUESTED** [review]
Don't forget about the tests

**Pull request closed without merging**